_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,411 @@
+
+## Configure authentication with OpenID Connect and Okta
+
+One more configuration step before running the architecture locally, let's configure Okta for authentication.
+
+{% include setup/cli.md type="jhipster" %}
+
+The settings from the generated `.okta.env` must be added to the `k8s/registry-k8s/application-configmap.yml`:
+
+```yml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+```
+
+Enable the OIDC authentication in the `jhipster-registry` service by adding the `oauth2` profile in the `k8s/registry-k8s/jhipster-registry.yml` file:
+
+```yml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+## Run locally with Minikube
+
+Install [Minikube](https://minikube.sigs.k8s.io/docs/start/) and [`kubectl`](https://kubernetes.io/docs/tasks/tools/).
+
+For Minkube, you will need at least 2 CPUs. Start MiniKube with your number of CPUs:
+
+```bash
+cd k8s
+minikube --cpus <ncpu> start
+```
+Minikube will log the Kubernetes and Docker versions on start:
+
+```
+Preparing Kubernetes v1.23.3 on Docker 20.10.12 ...
+```
+
+For the `store-mongodb` deployment to work, the property `Service.spec.publishNotReadyAddresses` was required, instead of the annotation `service.alpha.kubernetes.io/tolerate-unready-endpoints`, as the latter was deprecated in [Kubernetes release 1.11](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.11.md#kubernetes-111-release-notes).
+
+In the k8s folder, edit `store-mongodb.yml` and add the `publishNotReadyAddresses: true` property to the `spec`:
+
+```yml
+# Headless service for DNS record
+apiVersion: v1
+kind: Service
+metadata:
+  name: store-mongodb
+  namespace: demo
+spec:
+  type: ClusterIP
+  clusterIP: None
+  publishNotReadyAddresses: true
+  ports:
+    - name: peer
+      port: 27017
+  selector:
+    app: store-mongodb
+```
+
+Then deploy the application to Minikube, in the `k8s` directory, run:
+
+```bash
+./kubectl-apply.sh -f
+```
+Now is a good time to install [k9s](https://k9scli.io/topics/install/), a terminal-based UI to interact with Kubernetes clusters. Then run k9s with:
+
+```bash
+k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-pods.png alt:"k9s UI" width:"800" %}{: .center-image }
+
+Check out the [commands list](https://k9scli.io/topics/commands/), some useful ones are:
+- `:namespace`: show all available namespaces
+- `:pods`: show all available pods
+You can navigate the pods with `ENTER` and go back with `ESC` keys.
+
+Set up port-forwarding for the JHipster Registry.
+
+```bash
+kubectl port-forward svc/jhipster-registry -n demo 8761
```

Comment:
I get:

```
Error from server (NotFound): namespaces "demo" not found
```

Reply:
```
$ kubectl get pods
```

Results in:

```
NAME                                  READY   STATUS                  RESTARTS   AGE
blog-676dd59b64-5srkb                 0/1     Running                 2          7m58s
blog-neo4j-0                          1/1     Running                 0          7m58s
gateway-66558d6fb9-6jlv4              0/1     Running                 2          7m59s
gateway-postgresql-74dbc8c987-4mpq2   1/1     Running                 0          7m59s
jhipster-registry-0                   1/1     Running                 0          7m59s
jhipster-registry-1                   1/1     Running                 0          7m51s
store-7c9f858ccc-z9b2p                0/1     Init:0/1                0          7m58s
store-mongodb-0                       0/1     Init:CrashLoopBackOff   6          7m58s
```

Reply:
So your pods are not in the namespace `demo` when using the M1?

Reply:
Yes, you are correct. This is not an issue anymore.